plugin/entrySchema.go

```diff
@@ -67,25 +66,24 @@ type EntrySchema struct {
 	entrySchema
 	metaAttributeSchemaObj interface{}
 	metadataSchemaObj      interface{}
-	partialSchema          bool
 }
 
 // NewEntrySchema returns a new EntrySchema object with the specified label.
 //
 // NOTE: If your entry's a singleton, then the label should match the entry's
 // name, i.e. the name that's passed into plugin.NewEntry.
 func NewEntrySchema(e Entry, label string) *EntrySchema {
```

Comment:
Note that the changes here mean we no longer need to return `EntrySchema` pointers. I chose not to do so here b/c it makes the builder pattern slightly more complicated. Specifically, plugin authors would need to do something like
```
s := plugin.NewEntrySchema(...)
return s.SetLabel(...).IsSingleton()
``` 
and if they ever want to re-use their schema after building it, then they'd need to remember to set the schema to the returned value. So something like
```
s := NewEntrySchema(e, "bar")
s.SetLabel("foo")
print(s.Label)
```
would return "bar". However, doing `s = s.SetLabel("foo")` would return "foo".

So if possible, I'd still like to return pointers. Pointers or no pointers, I think we'd still need to check TypeIDs + ChildSchemas(), esp. since the latter could still return nil (i.e. some validation will always be needed). So may as well return pointers if it's not too much of a performance hit.